app/graphql/mutations/update_project_permissions.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class Mutations::UpdateProjectPermissions < Mutations::BaseMutation
+  argument :project_permissions, [Types::ProjectPermissionsAttributes], required: true
+
+  field :errors, [String], null: false
+
+  def resolve(project_permissions:)
+    projects = {}
+    users = {}
+
+    # This should be an all or nothing update
+    ActiveRecord::Base.transaction do
+      project_permissions.each do |project_permission|
+        project_string_key = project_permission.project_string_key
+        user_id = project_permission.user_id
+        permission_actions = project_permission.permissions
+
+        # Cache projects and users for future lookups
+        project = projects[project_string_key] ||= Project.find_by!(string_key: project_string_key)
```

Comment:
Yes!  That's something that I had in my early plans for Hyacinth 3.  Maybe caching projects, users, etc. in Redis and busting that cache whenever any users/projects are updates (since it's an infrequent thing).